src/app/screens/LNURLAuth/index.tsx

```diff
@@ -4,28 +4,40 @@ import ContentMessage from "@components/ContentMessage";
 import PublisherCard from "@components/PublisherCard";
 import { MouseEvent } from "react";
 import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
 import { useNavigationState } from "~/app/hooks/useNavigationState";
 import { USER_REJECTED_ERROR } from "~/common/constants";
 import api from "~/common/lib/api";
 import msg from "~/common/lib/msg";
 import type { LNURLAuthServiceResponse } from "~/types";
 
 function LNURLAuth() {
+  const navigate = useNavigate();
+
   const navState = useNavigationState();
-  const details = navState.args?.lnurlDetails as LNURLAuthServiceResponse;
+  const details = navState.args?.lnurlDetails as LNURLAuthServiceResponse & {
+    url: string;
+  };
 
   const { t } = useTranslation("components", {
     keyPrefix: "confirmOrCancel",
   });
 
   async function confirm() {
-    if (navState.isPrompt) {
-      return await msg.reply({
+    const response = await api.lnurlAuth({
```

Comment:
this should probably be split and no prompt/popup custom options should be passed to the action.

```
const response = await api.lnurlAuth({...});
await api.addOrUpdateAllowance({host: origin.host, lnurlAuth: true});
if (navState.isPrompt) {
  return await msg.reply(response);
} else {
  navigate(-1);
}